### ISSUE TITLE
Add a new <matchers/> subelement <foreignKeys/>

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14993]
In order to be able to name methods created from foreign keys (mainly for implicit joins), it would be useful to have a new `<matchers/>` subelement called `<foreignKeys/>`, with the following structure:

```xml
<foreignKeys>
  <foreignKey>

    <!-- Match unqualified or qualified table names. If left empty, this matcher applies to all tables. -->
    <expression>MY_TABLE</expression>

    <!-- These elements influence the naming of a generated methods. -->
    <methodName> see below MatcherRule specification </methodName>
    <methodNameInverse> see below MatcherRule specification </methodNameInverse>
    <methodNameManyToMany> see below MatcherRule specification </methodNameManyToMany>
  </foreignKey>
</foreignKeys>
```

For more strategic naming, it is probably better to use a programmatic generator strategy:
https://www.jooq.org/doc/dev/manual/code-generation/codegen-generatorstrategy/

But for quick changes in case an ambiguity needs to be resolved, this may suffice.

**Comment by lukaseder** [comment]
Given that the matcher is completely based on the foreign key name, this may not be such a useful feature on its own.

Perhaps, the `MatcherRule` should introduce a few variables, such as:

- `$catalog`
- `$schema`
- `$table`

Etc.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.19
